### DISCUSSION TITLE
PathTypeError with PlanetScope L2 composite: What am I missing?

### DISCUSSION THREAD

**Discussion by brobran** [sertit/eoreader#90] [Q&A] [answered]
Hey there, 
I am currently trying to read some L2 PlanetScope data with the specific metadata and use it in the xarray format for further processing.
The data comes in a folder that contains the `manifest.json` file and everything else (.tifs etc.). 

This is my current code:
``` Python 
from eoreader.reader import Reader
import os   
 
prod_folder = os.path.join("C:/", "Users", "sungg1024", "Desktop")
prod_path = os.path.join(prod_folder, "Ostersee24_08_2022_psscene_analytic_8b_sr_udm2")
reader = Reader()
prod = reader.open(prod_path)
```

This results in the error:
```
AnyPathTypeError: Invalid input for both CloudPath and Path. CloudPath exception: InvalidPrefixError("Path [] does not begin with a known prefix ['az://', 's3://', 'gs://'].") Path exception: TypeError('expected str, bytes or os.PathLike object, not list')
```

This is what is contained in the Planetscope file: 
![grafik](https://github.com/sertit/eoreader/assets/133111375/1de348dd-06b7-40d0-a2e5-0d7b5021da8e)
The tifs and everything else: 
![grafik](https://github.com/sertit/eoreader/assets/133111375/fb073a47-2e49-4a26-be45-c57d0a442863)

Is it maybe the name of the file that is causing an issue?

**Comment by remi-braun** [comment]
Hello !

I’m currently on holidays so I’ll try to debug it on the phone 😅

I think there is an issue about the provided path. It crashes before any attempt to read the product
What does `os.path.exists(prod_path)` return ?

**Reply by brobran** [reply]
Hi thanks a lot for the fast answer!! :D 

It returns `True`. I also tried to put the path directly into the open function like this  `reader.open("C:/Path/to/file")` but the error message remained the same.

**Reply by remi-braun** [reply]
Hmmm ok this is weird 😰

Note that mosaicked products are really tricky : eoreader will try to compute a VRT with all your tiles. 

I’m not sure if this product is recognized or not.

Can you send me the whole traceback ?

**Reply by remi-braun** [reply]
Hmmm wait a second, this composite product may not be handled! Can you show me the name of all the files contained in this folder ?

**Reply by brobran** [reply]
The traceback: 
```
Traceback (most recent call last):

  File ~\Anaconda3\envs\test_ma\Lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File c:\users\sungg1024\nextcloud\martini_br\code\planetdata.py:16
    prod = reader.open(prod_path)

  File ~\Anaconda3\envs\test_ma\Lib\site-packages\eoreader\reader.py:541 in open
    prod = class_(

  File ~\Anaconda3\envs\test_ma\Lib\site-packages\eoreader\products\optical\planet_product.py:147 in __init__
    super().__init__(product_path, archive_path, output_path, remove_tmp, **kwargs)

  File ~\Anaconda3\envs\test_ma\Lib\site-packages\eoreader\products\optical\optical_product.py:117 in __init__
    super().__init__(product_path, archive_path, output_path, remove_tmp, **kwargs)

  File ~\Anaconda3\envs\test_ma\Lib\site-packages\eoreader\products\product.py:253 in __init__
    self._post_init(**kwargs)

  File ~\Anaconda3\envs\test_ma\Lib\site-packages\eoreader\products\optical\pla_product.py:208 in _post_init
    super()._post_init(**kwargs)

  File ~\Anaconda3\envs\test_ma\Lib\site-packages\eoreader\products\optical\planet_product.py:203 in _post_init
    band_name = files.get_filename(self.get_default_band_path()).upper().split("_")

  File ~\Anaconda3\envs\test_ma\Lib\site-packages\sertit\files.py:701 in get_filename
    file_path = AnyPath(file_path)

  File ~\Anaconda3\envs\test_ma\Lib\site-packages\cloudpathlib\anypath.py:27 in __new__
    raise AnyPathTypeError(

AnyPathTypeError: Invalid input for both CloudPath and Path. CloudPath exception: InvalidPrefixError("Path [] does not begin with a known prefix ['az://', 's3://', 'gs://'].") Path exception: TypeError('expected str, bytes or os.PathLike object, not list')
``` 

The files that are in the folder are all in the screenshots of the original comment above.

**Reply by remi-braun** [reply]
Yeah ok this is it, the composite product is not handled.

I never used one, can you describe it for me please ? Is it orthorectified ? If so, a workaround could be to use a CustomProduct

If this composite product describes spectral bands or indices, you can open an issue demanding for its handling, I may have time to do it for the next release (with maybe your help 😉)

**Reply by brobran** [reply]
Okay I already suspected that the composite might be an issue. 
But then I'll just use the basic L1 PlanetScope non composite products, that should fix the issue I think.

Thank you a lot for your help!

**Comment by remi-braun** [comment, marked as answer]
TLDR: Composite products are not (yet) handled 😅
